server/src/main/java/org/spine3/server/command/CommandBus.java

```diff
@@ -215,13 +222,34 @@ private void handleUnsupported(Command command, StreamObserver<Response> respons
         responseObserver.onError(invalidArgumentWithCause(unsupported));
     }
 
+    private void scheduleAndStore(Command command, StreamObserver<Response> responseObserver) {
+        if (scheduler == null) {
+            final StatusRuntimeException noSchedulerException = failedPreconditionNoScheduler();
+            responseObserver.onError(noSchedulerException);
+            commandStore.store(command, noSchedulerException);
+            return;
+        }
+        scheduler.schedule(command);
+        commandStore.store(command, SCHEDULED);
+        responseObserver.onNext(Responses.ok());
+        responseObserver.onCompleted();
+    }
+
     private static StatusRuntimeException invalidArgumentWithCause(Exception exception) {
         final StatusRuntimeException result = Status.INVALID_ARGUMENT
                 .withCause(exception)
                 .asRuntimeException();
         return result;
     }
 
+    private static StatusRuntimeException failedPreconditionNoScheduler() {
```

Comment:
I would keep the term `precondition` to parameter checking. Here we talk about illegal state.